driver/src/main/java/org/neo4j/driver/internal/async/pool/ConnectionPoolImpl.java

```diff
@@ -226,6 +230,13 @@ private void processAcquisitionError( BoltServerAddress serverAddress, Throwable
                         "Unable to acquire connection from the pool within configured maximum time of " +
                         settings.connectionAcquisitionTimeout() + "ms" );
             }
+            else if ( pool.isClosed() )
+            {
+                // There is a race condition where a thread tries to acquire a connection while the pool is closed by another concurrent thread.
+                // Treat as failed to obtain connection for a direct driver. For a routing driver, this error should be retried.
+                throw new ServiceUnavailableException( format( "Connection pool for server %s is closed while acquiring a connection. ", serverAddress ),
```

Comment:
The extra space after `a connection.` in the exception message seems to be a typo.